library/Freckle/App/Kafka/Producer.hs

```diff
@@ -123,25 +125,27 @@ produceKeyedOn prTopic values keyF = traced $ do
   view kafkaProducerPoolL >>= \case
     NullKafkaProducerPool -> pure ()
     KafkaProducerPool producerPool -> do
-      errors <-
-        liftIO $
-          Pool.withResource producerPool $ \producer ->
-            produceMessageBatch producer $
-              toList $
-                mkProducerRecord <$> values
-      unless (null errors) $
-        logErrorNS "kafka" $
-          "Failed to send events" :# ["errors" .= fmap (tshow . snd) errors]
+      errors <- liftIO $ Pool.withResource producerPool $ \producer ->
+        Conduit.runConduit
```

Comment:
What's up with using conduit here instead of a plain `traverse`?

Reply:
I find it easier to write.

Reply:
I didn't even notice this. I read your blog about this, and I'm into it in spirit. Particulary when you're doing some version of concat-map-maybe like this. But I dunno, the alternative is a quite a bit more accessible IMO:

```hs
errors <- liftIO $ Pool.withResource producerPool $ \producer ->
  fmap catMaybes $ traverse (produceMessage producer . mkProducerRecord) values
```

Maybe not qualifiying the `Conduit.` would make me feel differently.

Reply:
The reason I ask is that my hackles go up a bit when I see conduit, and I assume that we needed it for some sort of constant space style processing. 

I agree that `fmap catMaybes . traverse` is a bit simpler for me to read at least.